pkg/generators/markers.go

```diff
@@ -0,0 +1,273 @@
+package generators
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"k8s.io/gengo/types"
+)
+
+// Subset of spec.SchemaProps that may be placed into a marker comment
+type ValueValidations struct {
+	Maximum          *float64 `json:"maximum,omitempty"`
+	ExclusiveMaximum *bool    `json:"exclusiveMaximum,omitempty"`
+	Minimum          *float64 `json:"minimum,omitempty"`
+	ExclusiveMinimum *bool    `json:"exclusiveMinimum,omitempty"`
+	MaxLength        *int64   `json:"maxLength,omitempty"`
+	MinLength        *int64   `json:"minLength,omitempty"`
+	Pattern          *string  `json:"pattern,omitempty"`
+	MaxItems         *int64   `json:"maxItems,omitempty"`
+	MinItems         *int64   `json:"minItems,omitempty"`
+	UniqueItems      *bool    `json:"uniqueItems,omitempty"`
+	MultipleOf       *float64 `json:"multipleOf,omitempty"`
+	MaxProperties    *int64   `json:"maxProperties,omitempty"`
+	MinProperties    *int64   `json:"minProperties,omitempty"`
+}
+
+type CommentTags struct {
+	ValueValidations
+
+	// Future markers can all be parsed into this centralized struct...
```

Comment:
I like that a this struct is a good place to move our existing tags to over time rather than have an ever-expanding piecemeal solution